.buildkite/pipeline.yml

```diff
@@ -51,8 +51,44 @@ steps:
           cache-from:
             - node-maze-runner:855461928731.dkr.ecr.us-west-1.amazonaws.com/js:ci-${BUILDKITE_BRANCH}
 
+  - label: ':docker: Build expo maze runner image'
+    plugins:
+      - docker-compose#v2.6.0:
+          build:
+            - expo-maze-runner
+          image-repository: 855461928731.dkr.ecr.us-west-1.amazonaws.com/js
+          cache-from:
+            - expo-maze-runner:855461928731.dkr.ecr.us-west-1.amazonaws.com/js:ci-${BUILDKITE_BRANCH}
+      - docker-compose#v2.6.0:
+          push:
+            - expo-maze-runner:855461928731.dkr.ecr.us-west-1.amazonaws.com/js:ci-${BUILDKITE_BRANCH}
+            - expo-maze-runner:855461928731.dkr.ecr.us-west-1.amazonaws.com/js:ci-latest
+
   - wait
 
+  - label:  ':docker: Publish expo app'
```

Comment:
This should have its own build step that caches properly as it takes 15 minutes! Currently you are caching from ci but it doesnt share any layers from ci because the first layer has differences - one is `RUN apk add --update bash git` and the other is `RUN apk add --update bash`. We should try to make the first few steps in expo-publisher the same as ci to make that cache reusable and only fall back to that if we dont have a pre-existing cache for the expo-publisher image on this `BUILDKITE_BRANCH`

Reply:
I've split it between a build step that runs in the same stage as building the maze images (that should run in approximately the same time as the node-maze image), and a run step in the same place as previous.  I'll run it a couple of times to check the caching, and see if there's a speed increase there.